arch/arm64/src/Toolchain.defs

```diff
@@ -183,13 +183,7 @@ endif
 # Workaround for GCC-12.2 linker warning
 
 ifeq ($(CONFIG_ARCH_TOOLCHAIN_GNU),y)
-  ifeq ($(GCCVER),)
-    export GCCVER := $(shell $(CC) --version | grep gcc | sed -E "s/.* ([0-9]+\.[0-9]+).*/\1/" | cut -d'.' -f1)
-  endif
-
-  ifeq ($(GCCVER),12)
```

Comment:
The `eq` here is wrong obviously and should be greater equal, but I think that main intension here was to add new flag that appeared starting from GCC 12.

BTW, you remove the condition, but comment above `# Workaround for GCC-12.2 linker warning` is still there ;)